azafea/event_processors/endless/country.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2020 - Endless OS LLC
+#
+# This file is part of Azafea
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from typing import List
+
+COUNTRIES = {
+    'ABW': 'AW', 'AGO': 'AO', 'ALA': 'AX', 'AND': 'AD', 'ARE': 'AE', 'ARM': 'AM', 'ATA': 'AQ',
+    'ATF': 'TF', 'ATG': 'AG', 'AUT': 'AT', 'BDI': 'BI', 'BEN': 'BJ', 'BES': 'BQ', 'BGD': 'BD',
+    'BHS': 'BS', 'BIH': 'BA', 'BLR': 'BY', 'BLZ': 'BZ', 'BRB': 'BB', 'BRN': 'BN', 'CAF': 'CF',
+    'CHL': 'CL', 'CHN': 'CN', 'COD': 'CD', 'COG': 'CG', 'COK': 'CK', 'COM': 'KM', 'CPV': 'CV',
+    'CUW': 'CW', 'CYM': 'KY', 'DNK': 'DK', 'ESH': 'EH', 'EST': 'EE', 'FLK': 'FK', 'FRO': 'FO',
+    'FSM': 'FM', 'GIN': 'GN', 'GLP': 'GP', 'GNB': 'GW', 'GNQ': 'GQ', 'GRD': 'GD', 'GRL': 'GL',
+    'GUF': 'GF', 'GUY': 'GY', 'IRL': 'IE', 'IRQ': 'IQ', 'ISR': 'IL', 'JAM': 'JM', 'KAZ': 'KZ',
+    'KOR': 'KR', 'LBR': 'LR', 'LBY': 'LY', 'MAC': 'MO', 'MAF': 'MF', 'MDG': 'MG', 'MDV': 'MV',
+    'MEX': 'MX', 'MLT': 'MT', 'MNE': 'ME', 'MNP': 'MP', 'MOZ': 'MZ', 'MTQ': 'MQ', 'MYT': 'YT',
+    'NIU': 'NU', 'PAK': 'PK', 'PCN': 'PN', 'PLW': 'PW', 'PNG': 'PG', 'POL': 'PL', 'PRK': 'KP',
+    'PRT': 'PT', 'PRY': 'PY', 'PYF': 'PF', 'SEN': 'SN', 'SGS': 'GS', 'SLB': 'SB', 'SLV': 'SV',
+    'SPM': 'PM', 'SRB': 'RS', 'SUR': 'SR', 'SVK': 'SK', 'SVN': 'SI', 'SWE': 'SE', 'SWZ': 'SZ',
+    'SYC': 'SC', 'TCD': 'TD', 'TKM': 'TM', 'TUN': 'TN', 'TUR': 'TR', 'TUV': 'TV', 'UKR': 'UA',
+    'URY': 'UY', 'WLF': 'WF'}
+
+
+def transform_alpha_3_into_2(tablename: str) -> List[str]:
+    requests = [
+        f"UPDATE {tablename} SET country = '{alpha_2}' WHERE country = '{alpha_3}'"
+        for alpha_3, alpha_2 in COUNTRIES.items()]
+    requests.append(
+        f'UPDATE {tablename} SET country = substring(country FROM 1 FOR 2) '
+        'WHERE char_length(country) = 3')
```

Comment:
I still don't really understand why this is preferable than making `COUNTRIES` a complete mapping from alpha-3 to alpha-2 for all known countries. (I believe that you have checked, of course!)

Reply:
I can do this if you prefer.

Reply:
It's a one-time migration so if this works, it's probably not worth spending much more time on it, but I would be more comfortable with this line removed and the COUNTRIES table updated to include all countries, yes.